app/controllers/api/account.php

```diff
@@ -3821,6 +3821,7 @@
             $mfaRecoveryCodes = $user->getAttribute('mfaRecoveryCodes', []);
             if (in_array($otp, $mfaRecoveryCodes)) {
                 $mfaRecoveryCodes = array_diff($mfaRecoveryCodes, [$otp]);
+                $mfaRecoveryCodes = array_values($mfaRecoveryCodes);
```

Comment:
when/how is the used OTP removed?

Reply:
Array_dif is removing it. It's done in 2 places. Few lines below, we are updating users document.